### DISCUSSION TITLE
3.3.7

### DISCUSSION THREAD

**Discussion by mathiascode** [nicotine-plus/nicotine-plus#3225] [Releases]
### Changes

 * Grey out ignored usernames in room user lists
 * Automatically reconnect to the server when changing listening port
 * Always show number of upload slots on user profiles
 * Windows: Update Python from 3.11 to 3.12
 * macOS: Drop support for macOS Monterey due to reaching EOL

### Corrections

 * Fixed a crash when using file choosers in plugin settings
 * Fixed a crash when upgrading from Nicotine+ 1.4.1 and earlier
 * Fixed broken upload stats caused by Soulseek NS clients resuming >2 GB downloads
 * Fixed an issue where picking an emoji would clear existing text
 * Improved file transfer speeds in some cases
 * Windows: Reverted switch to Clang environment due to rare crashes

### Issues closed on GitHub

 * Banned and ignored users indicators ([#2100](https://github.com/nicotine-plus/nicotine-plus/issues/2100))
 * Allow N+ to update Port without restarting ([#2704](https://github.com/nicotine-plus/nicotine-plus/issues/2704))
 * Incorrect processing of passwords with special characters ([#3156](https://github.com/nicotine-plus/nicotine-plus/issues/3156))
 * 3.3.5 has same bug latest Test clients have about upload showing wrong upload ([#3162](https://github.com/nicotine-plus/nicotine-plus/issues/3162))
 * Cannot update nicotine+ ([#3167](https://github.com/nicotine-plus/nicotine-plus/issues/3167))
 * 3.3.6 crashes on launch on macOS due to broken rendering ([#3172](https://github.com/nicotine-plus/nicotine-plus/issues/3172))
 * GTK4-based app crashes with: TypeError: No means to translate argument or return value for 'GdkButtonEvent' ([#3176](https://github.com/nicotine-plus/nicotine-plus/issues/3176))
 * I get this error when i try to run Nicotine+ ([#3175](https://github.com/nicotine-plus/nicotine-plus/issues/3175))
 * A lot of dump files (gdbus) ([#3181](https://github.com/nicotine-plus/nicotine-plus/issues/3181))
 * Filechooser.py Window' object has no attribute 'widget' ([#3182](https://github.com/nicotine-plus/nicotine-plus/issues/3182))
 * \[3.3.7.dev1\] Windows Nightly builds libgcc_s_seh-1.dll is missing ([#3183](https://github.com/nicotine-plus/nicotine-plus/issues/3183))
 * \[Bug\]: Transfer Statistics Total Uploaded Size jumps to 32 EiB ([#3185](https://github.com/nicotine-plus/nicotine-plus/issues/3185))
 * Picking an emoji clears text in chat entry, rather than being appended to it ([#3197](https://github.com/nicotine-plus/nicotine-plus/issues/3197))
 * User logs off but items in queue slowly tick away from saying queued to logged off ([#3209](https://github.com/nicotine-plus/nicotine-plus/issues/3209))
 * Nicotine+ not working on Windows 10 since v3.3.6 ([#3210](https://github.com/nicotine-plus/nicotine-plus/issues/3210))

<hr /><em>This discussion was created from the release <a href='https://github.com/nicotine-plus/nicotine-plus/releases/tag/3.3.7'>3.3.7</a>.</em>

**Comment by mlnl221** [comment]
thanks for the release! the reconnect works perfectly and my vpn plugin will auto set the port now. thanks for the effort

I'm also getting way faster upload speed on certain peers, where it would normally upload 7MiB/s now I am getting upload speed of up to 40MiB/s! Great update.

This of course highly depends on the peer since I have 1GiB upload speed in my house and through my VPN.
